### PR TITLE
refactor(clp-package): Unify the metadata schema for JSON and IR streams.

### PR DESCRIPTION
**Pull request by haiqi96**
# Description
This PR is a continuation of work in #596 
The PR introduces three updates to the stream metadata so that both IR and JSON metadata share the same scheme:
1. remove file_split_id from IR metadata, since it is not used by the WebUI.
2. replace orig_file_id and archive_id with "stream_id". 
3. Rename "is_last_ir_chunk" to "is_last_chunk".

With the changes above, we are able to simplify the webui code in the same PR.

# Validation performed
Manually tested both CLP and CLP-S log viewing.
Manually verified that extracted stream metadata in the mongodb matches expectation

## Release Notes

- **New Features**
	- Enhanced metadata handling for extracted streams with updated identifiers.
	- Added a new `stream_id` property to the response of extraction jobs.

- **Bug Fixes**
	- Improved error handling for connection failures in result caching.
	- Enhanced error handling for file decompression scenarios.

- **Documentation**
	- Updated type definitions and return types for API calls related to stream extraction.
	- Clarified parameter documentation for the decompression methods.

These changes improve the overall functionality and reliability of the application, ensuring better alignment with the updated data structure.

### CODE REVIEW

File: components/core/src/clp/clo/OutputHandler.cpp [hunk 1]
```diff
@@ -100,7 +100,7 @@ ErrorCode ResultsCacheOutputHandler::flush() {
         try {
             m_results.emplace_back(std::move(bsoncxx::builder::basic::make_document(
                     bsoncxx::builder::basic::kvp(
-                            cResultsCacheKeys::OrigFileId,
+                            cResultsCacheKeys::SearchOutput::OrigFileId,
```
[the remark this answers is not in the document]
Comment (haiqi96) [reply components/core/src/clp/clo/OutputHandler.cpp:103]: This change is expected. This line is not for extracted stream metadata.

File: components/core/src/clp/clo/clo.cpp [hunk 2]
```diff
@@ -195,13 +195,9 @@ bool extract_ir(CommandLineArguments const& command_line_args) {
                             dest_ir_file_name
                     ),
                     bsoncxx::builder::basic::kvp(
-                            clp::clo::cResultsCacheKeys::OrigFileId,
+                            clp::clo::cResultsCacheKeys::IrOutput::StreamId,
```
[the remark this answers is not in the document]
Comment (haiqi96) [reply components/core/src/clp/clo/clo.cpp:198]: Those are fine.

**Review by gibber9809 — APPROVED** [review]
LGTM.

**Comment by gibber9809** [comment]
> LGTM.

I think for the title you may just need a space after the ":" character.

**Review by junhaoliao — APPROVED** [review]
the webUI changes lgtm

**Review by kirkrodrigues — APPROVED** [review]
For the PR title, how about:

```
refactor(clp-package): Unify the metadata schema for JSON and IR streams.
```

**Pull request merged**